### PR TITLE
[Draft] OpenSearch SQL autosuggest with Antlr

### PR DESCRIPTION
**Pull request by mengweieric**
### Description

currently on cleanup, tests and some minor adjustments.

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -30,3 +30,7 @@ target
 /packages/osd-test/src/functional_test_runner/lib/config/__tests__/fixtures/
 /packages/osd-ui-framework/dist
 /packages/osd-ui-shared-deps/flot_charts
+
+# antlr overrides
+/src/plugins/data/public/antlr/opensearch_sql/.generated/*
```
Comment (mengweieric) [inline .eslintignore:35]: ignore setup here and below in eslintrc.js doesn't seem to work

File: package.json [hunk 2]
```diff
@@ -11,7 +11,7 @@
     "dashboarding"
   ],
   "private": true,
-  "version": "3.0.0",
+  "version": "2.14.0",
```
Comment (joshuali925) [inline package.json:14]: revert this

File: package.json [hunk 3]
```diff
@@ -84,7 +84,8 @@
     "cypress:run-without-security": "env TZ=America/Los_Angeles NO_COLOR=1 cypress run --headless --env SECURITY_ENABLED=false",
     "cypress:run-with-security": "env TZ=America/Los_Angeles NO_COLOR=1 cypress run --headless --env SECURITY_ENABLED=true,openSearchUrl=https://localhost:9200,WAIT_FOR_LOADER_BUFFER_MS=500",
     "osd:ciGroup10": "echo \"dashboard_sanity_test_spec.js\"",
-    "osd:ciGroup11": "echo \"apps/vis_builder/*.js\""
+    "osd:ciGroup11": "echo \"apps/vis_builder/*.js\"",
+    "generate:opensearchsqlantlr": "./node_modules/antlr4ng-cli/index.js -Dlanguage=TypeScript -o ./src/plugins/data/public/antlr/opensearch_sql/generated -visitor -no-listener -Xexact-output-dir /home/ubuntu/ws/OpenSearch-Dashboards/src/plugins/data/public/antlr/opensearch_sql/grammar/OpenSearchSQLLexer.g4 /home/ubuntu/ws/OpenSearch-Dashboards/src/plugins/data/public/antlr/opensearch_sql/grammar/OpenSearchSQLParser.g4"
```
Comment (joshuali925) [inline package.json:88]: convert the absolute paths to relative?

File: src/plugins/data/public/ui/query_editor/suggestion_helpers.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Any modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
```
Comment (joshuali925) [inline src/plugins/data/public/ui/query_editor/suggestion_helpers.tsx:9]: use the shorter version?

File: src/plugins/data/public/ui/query_editor/suggestion_helpers.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+ */
+
+export const formatSuggestions = (suggestions) => {
```
Comment (joshuali925) [inline src/plugins/data/public/ui/query_editor/suggestion_helpers.tsx:12]: type?

File: src/plugins/data/public/ui/query_string_input/query_string_input.tsx [hunk 6]
```diff
@@ -191,8 +189,7 @@ export default class QueryStringInputUI extends Component<Props, State> {
           selectionEnd,
           signal: this.abortController.signal,
         })) || [];
-
-      return [...suggestions, ...recentSearchSuggestions];
+      return [...suggestions];
```
Comment (joshuali925) [inline src/plugins/data/public/ui/query_string_input/query_string_input.tsx:192]: i thought autocomplete only applies to the query editor, does it need to change query string input as well?

File: src/plugins/data/public/ui/query_editor/language_switcher.tsx [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+function mapExternalLanguageToOptions(language: string) {
+  return {
+    label: language,
+    value: language,
+  };
+}
+
+export function QueryLanguageSwitcher(props: Props) {
```
Comment (joshuali925) [inline src/plugins/data/public/ui/query_editor/language_switcher.tsx:25]: there is already a query language selector, how is this different?

**Review by joshuali925 — COMMENTED** [review]
some minor questions, will review completion logic later

File: package.json (hunk 2 above)
Comment (mengweieric) [reply package.json:14]: yea this is under clean up radar.

File: package.json (hunk 3 above)
Comment (mengweieric) [reply package.json:88]: yes, it's also included in clean up tasks

File: src/plugins/data/public/ui/query_editor/suggestion_helpers.tsx (hunk 5 above)
Comment (mengweieric) [reply src/plugins/data/public/ui/query_editor/suggestion_helpers.tsx:12]: yea same cleanup tasks

File: src/plugins/data/public/ui/query_editor/language_switcher.tsx (hunk 7 above)
Comment (mengweieric) [reply src/plugins/data/public/ui/query_editor/language_switcher.tsx:25]: this is not added by me but seems like appeared after the rebase..will check

File: src/plugins/data/public/ui/query_string_input/query_string_input.tsx (hunk 6 above)
Comment (mengweieric) [reply src/plugins/data/public/ui/query_string_input/query_string_input.tsx:192]: testing changes, also already been included in cleanup

**Comment by mengweieric** [comment]
switching branch, closing this one.

**Pull request closed without merging**